src/main/java/com/google/sps/servlets/DirectionsServlet.java

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.common.collect.ImmutableList;
+import com.google.gson.Gson;
+import com.google.sps.exceptions.DirectionsException;
+import com.google.sps.model.DirectionsClient;
+import com.google.sps.model.DirectionsClientFactory;
+import com.google.sps.model.DirectionsClientImpl;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Serves key information from optimizing between addresses. */
+@WebServlet("/directions")
+public class DirectionsServlet extends HttpServlet {
+
+  private final DirectionsClientFactory directionsClientFactory;
+  private final String apiKey;
+  private final String origin;
+  private final String destination;
+  private final List<String> waypoints;
+
+  /** Construct servlet with default DirectionsClient. */
+  public DirectionsServlet() {
+    directionsClientFactory = new DirectionsClientImpl.Factory();
+    apiKey = "Fake API Key";
```

Comment:
FYI that this key will get passed to `DirectionsClientImpl` and the real Directions API, and it won't work.

Reply:
Echoing from our discussion on chat - I definitely don't want to be merging unusable code into master. Will start using KeyProvider in this PR to avoid having the API key exposed and have the API working at the same time.